### PR TITLE
Release workflow improvements [main]

### PR DESCRIPTION
**Pull request by gururajsh**
## Description of the Change

This change is to update our release process. It also
- Removes secrets environment usage from workflows
- Uploads artifacts to github release
- Adds release notes template
- Creates release notes for github release
- Updates CLAW with new release version

## Why Is This PR Valuable?

It improves time needed to do a CLI release.

## Applicable Issues

None

## How Urgent Is The Change?

Is the change urgent? If so, explain why it is time-sensitive.

## Other Relevant Parties

Who else is affected by the change?

### CODE REVIEW

File: .github/release/release-notes-template copy.txt
Comment (a-b) [inline .github/release/release-notes-template copy.txt:1]: Is suffix ` copy.txt` intended?

File: .github/workflows/release-build-sign-upload.yml [hunk 1]
```diff
@@ -80,10 +80,8 @@ jobs:
     name: Setup
     # needs: test
     runs-on: ubuntu-latest
-    if: ${{ github.action_repository != 'cloudfoundry/cli' }}
```
Comment (a-b) [inline .github/workflows/release-build-sign-upload.yml:83]: 👍

File: .github/release/release-notes-template copy.txt
Comment (a-b) [reply .github/release/release-notes-template copy.txt:1]: Would you like us to keep this suffix moving forward?

Comment (gururajsh) [reply .github/release/release-notes-template copy.txt:1]: Nope, will fix it

Comment (gururajsh) [reply .github/release/release-notes-template copy.txt:1]: Fixed. Thanks for catching this

**Pull request merged**